modular_skyrat/modules/modular_implants/code/soulcatcher/soulcatcher_items.dm

```diff
@@ -0,0 +1,143 @@
+	name = "'Evoker'-Type RSD"
+	desc = "The 'Evoker'-Type RSD, or Resonance Simulation Device, is a sort of 'Soulcatcher' instrument that's been designated for handheld usage. These RSDs were designed with the Medical field in mind, a tool meant to offer comfort to the temporarily-departed while their bodies are being repaired, healed, or produced. The Evoker is essentially a very specialized handheld NIF, still using the same nanomachinery for the software and hardware. This careful instrument is able to host a virtual space for a great number of Engrams for an essentially indefinite amount of time in an unlimited variety of simulations, even able to transfer them to and from a NIF. However, it's best Medical practice to not lollygag."
+	icon = 'modular_skyrat/modules/modular_implants/icons/obj/devices.dmi'
+	icon_state = "soulcatcher-device"
+	inhand_icon_state = "electronic"
+	lefthand_file = 'icons/mob/inhands/items/devices_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/items/devices_righthand.dmi'
+	/// What soulcatcher datum is associated with this item?
+	var/datum/component/soulcatcher/linked_soulcatcher
+
+/obj/item/soulcatcher_item/attack_self(mob/user, modifiers)
+	linked_soulcatcher.ui_interact(user)
+
+/obj/item/soulcatcher_item/New(loc, ...)
+	. = ..()
+	linked_soulcatcher = AddComponent(/datum/component/soulcatcher)
+
+/obj/item/soulcatcher_item/Destroy(force)
+	if(linked_soulcatcher)
+		qdel(linked_soulcatcher)
+
+	return ..()
+
+/obj/item/soulcatcher_item/attack(mob/living/target_mob, mob/living/user, params)
+	if(!target_mob)
+		return ..()
+
+	var/datum/component/previous_body/body_component = target_mob.GetComponent(/datum/component/previous_body)
+	if(body_component)
+		scan_body(body_component, user)
+		return TRUE
+
+	if(!target_mob.mind || !target_mob.ckey)
+		to_chat(user, span_warning("You are unable to remove a mind from an empty body."))
+		return FALSE
+
+	if(target_mob.stat == DEAD) //We can temporarily store souls of dead mobs.
+		target_mob.ghostize(TRUE) //Incase they are staying in the body.
+		var/mob/dead/observer/target_ghost = target_mob.get_ghost(TRUE, TRUE)
+		if(!target_ghost)
+			to_chat(user, span_warning("You are unable to get the soul of [target_mob]!"))
+			return FALSE
+
+		var/datum/soulcatcher_room/target_room = tgui_input_list(user, "Chose a room to send [target_mob]'s soul to.", name, linked_soulcatcher.soulcatcher_rooms)
+		if(!target_room)
+			return FALSE
+
+		if(tgui_alert(target_ghost, "[user] wants to transfer you to a soulcatcher, do you accept?", name, list("Yes", "No"), autofocus = TRUE) != "Yes")
+			to_chat(user, span_warning("[target_mob] doesn't seem to want to enter."))
+			return FALSE
+
+		if(!target_room.add_soul_from_ghost(target_ghost))
+			return FALSE
+
+		body_component = target_mob.GetComponent(/datum/component/previous_body)
+		if(!body_component)
+			return FALSE
+
+		log_admin("[user] used [src] to put [target_mob]'s mind into a soulcatcher.")
```

Comment:
Stuff like this should probably also have log coordinates (I don't remember how you do them, but I'm almost certain there's a helper/macro for it).